Sources/TSCBasic/misc.swift

```diff
@@ -347,8 +347,14 @@ extension SystemError: CustomStringConvertible {
         switch self {
         case .chdir(let errno, let path):
             return "chdir error: \(strerror(errno)): \(path)"
-        case .close(let errno):
-            return "close error: \(strerror(errno))"
+        case .close(let err):
+            let errorMessage: String
+            if err == -1 { // if the return code is -1, we need to consult the global `errno`
+                errorMessage = strerror(errno)
```

Comment:
you need to copy this string immediately or else it's UB and you might get garbage or worse:

man strerror:

> BUGS
>      For unknown error numbers, the strerror() function will return its result
>     in a static buffer which may be overwritten by subsequent calls.

So you want to save a `String(cString: strerror(errno)) + " (\(errno))"` or something. I'd say let's not lose the actual errno number.

Reply:
Looks like we are basically doing that, the `strerror` here is not the libc function, but a local one https://github.com/apple/swift-tools-support-core/blob/main/Sources/TSCBasic/misc.swift#L320-L345